### PR TITLE
Input/output byte metrics for HTTP destination and non-blocking sources/destinations

### PR DESCRIPTION
**Pull request by MrAnno**
This PR introduces byte-based input/output metrics for all non-blocking sources and destinations and for the HTTP destination.

On 32-bit architectures, metrics use `SCU_KIB` and cache smaller values in order to represent values larger than 4 GiB.

These metrics show the serialized message sizes (protocol-specific header/framing length is not included).

The incoming message length is also available as a hard macro: `$RAWMSG_SIZE`.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng do stresstest

**Comment by kira-syslogng** [comment]
Kira-stress-test: Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/logsource.c [hunk 1]
```diff
@@ -457,6 +457,57 @@ _create_ack_tracker_if_not_exists(LogSource *self)
     }
 }
 
+static void
+_register_raw_bytes_stats(LogSource *self)
+{
```
Comment (alltilla) [inline lib/logsource.c:462]: As we talked about it IRL, this should be optional, so we do not create unused counters for Fetchers, etc.

File: lib/logmsg/logmsg.h [hunk 2]
```diff
@@ -478,6 +476,19 @@ void log_msg_tags_foreach(const LogMessage *self, LogMessageTagsForeachFunc call
 void log_msg_format_tags(const LogMessage *self, GString *result);
 void log_msg_format_matches(const LogMessage *self, GString *result);
 
+
+static inline void
+log_msg_set_recvd_rawmsg_size(LogMessage *self, guint32 size)
+{
+  self->recvd_rawmsg_size = size;
+}
```
Comment (alltilla) [inline lib/logmsg/logmsg.h:484]: As we talked about it IRL, we could make it, so the Source updates the input_event_bytes_total counter based on this field, so the descendant classes only need to call this setter, and not the stats update fn.

**Review by alltilla — COMMENTED** [review]
Cool feature!

File: lib/logmsg/logmsg.h (hunk 2 above)
Comment (MrAnno) [reply lib/logmsg/logmsg.h:484]: Fixed.

File: lib/logsource.c [hunk 3]
```diff
@@ -457,6 +457,57 @@ _create_ack_tracker_if_not_exists(LogSource *self)
...
+    return;
+
+  self->metrics.recvd_bytes_cache += b;
+
+  if (self->metrics.recvd_bytes_cache > 1024)
+    {
+      stats_counter_add(self->metrics.recvd_kbytes, self->metrics.recvd_bytes_cache / 1024);
+      self->metrics.recvd_bytes_cache %= 1024;
+    }
+}
```
[the remark this answers is not in the document]
Comment (MrAnno) [reply lib/logsource.c]: I've added this layer. Please take a look.

**Comment by MrAnno** [comment]
@kira-syslogng do stresstest

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng do stesstest

**Comment by MrAnno** [comment]
@kira-syslogng do stresstest

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Kira-stress-test: Build SUCCESS

File: lib/logsource.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (alltilla) [reply lib/logsource.c]: Cool! Thanks

**Review by alltilla — APPROVED** [review]

**Pull request merged**